### PR TITLE
perf(install): Avoid checking all files for unlink persisted data

### PR DESCRIPTION
**Pull request by younger-1**
#### Description

My commit involved two changes:

1. when uninstall or reset <app>, avoid checking all files for unlink persisted data in function `unlink_persist_data`. It will take a long time to check all the files when the <app> contains a great deal of files by itself or would produce many new file in the daily usage. For example: `miniconda3` and `anaconda3` kick in above two situation and would produce many new hard-link file when installing new environments.

2. when shim, add `-CommandType:Application` option to `git`. This is not that vital but it could cause problem if `git` is defined as a `powershell` function by user like me

#### How Has This Been Tested?

I reset all my scoop apps (every kinds of app from every kinds of bucket, 482 apps in total), and the whole process worked well.

#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by niheaven** [comment]
You should add a CHANGLOG entry, thanks. and I'm on vacation soon, so the reviewing time may be longer.

**Comment by younger-1** [comment]
Sorry for late replying, I will add it.

**Comment by younger-1** [comment]
Forget to mention after changelog updated 🥲 . All things looks right now. Could you merge this? @niheaven

**Review by niheaven — APPROVED** [review]

**Review by niheaven — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,10 @@
 - **scoop-cleanup:** Remove apps other than current version ([#4665](https://github.com/ScoopInstaller/Scoop/issues/4665))
 - **scoop-update:** Skip updating non git buckets ([#4670](https://github.com/ScoopInstaller/Scoop/issues/4670), [#4672](https://github.com/ScoopInstaller/Scoop/issues/4672))
 
+### Performance Improvements
+
+- **install:** Avoid checking all files for unlinking persisted data ([#4681](https://github.com/ScoopInstaller/Scoop/issues/4681))
```
Comment (rashil2000) [inline CHANGELOG.md:40]:
```suggestion
- **uninstall:** Avoid checking all files for unlinking persisted data ([#4681](https://github.com/ScoopInstaller/Scoop/issues/4681))
```

**Comment by rashil2000** [comment]
LGTM

**Review by niheaven — APPROVED** [review]

**Pull request merged**